pyteal/ast/abi/method_return.py

```diff
@@ -0,0 +1,36 @@
+from typing import TYPE_CHECKING, Tuple
+from . import BaseType
+from ...types import TealType
+from ...errors import TealInputError
+from .. import Expr, Log
+from ...ir import TealBlock, TealSimpleBlock, Op
+
+if TYPE_CHECKING:
+    from ...compiler import CompileOptions
+
+
+class MethodReturn(Expr):
+    def __init__(self, arg: BaseType):
+        super().__init__()
+        if not isinstance(arg, BaseType):
+            raise TealInputError(f"Expecting an ABI type argument but get {arg}")
+        self.arg = arg
+
+    def __teal__(self, options: "CompileOptions") -> Tuple[TealBlock, TealSimpleBlock]:
+        if options.version < Op.log.min_version:
+            raise TealInputError(
+                f"current version {options.version} is lower than log's min version {Op.log.min_version}"
+            )
+        return Log(self.arg.encode()).__teal__(options)
```

Comment:
Should this Concat with the RETURN_EVENT_SELECTOR ?

Reply:
oof I was rushing this on meeting, lots of mistakes my bad